### DISCUSSION TITLE
Increase column size of UserAttribute table

### DISCUSSION THREAD

**Discussion by kluangguy1** [keycloak/keycloak#13494] [Q&A] [answered]
Dear all,
I have a use case to capture the additional OIDC claim from an IDP and store it into the user attribute such that I can include it into the ID Token response which can be consumed by my application . I am defining a Identity Provider mappers to store the such claim.

however, the size of the claim exceeded the column size of "value" (varchar 255) in the user_attribute table. is it something i can perform by increasing such column size from 255 char to 4000 char? is there any impact to the keycloak?

Thanks

**Comment by sschu** [comment, marked as answer]
You can pretty safely just increase the column size. You should put a limit in there, as without limit, there might be problems with indexes. There are even ways to add custom migrations to Keycloak if you need to replicate that setup, but it's a bit hacky.

**Comment by cgeorgilakis** [comment]
I can not understand why Keycloak team does not increase User Attribute column at least at 4000 char. We also have User Attribute values that 255 chars is too little.
We use PostgreSQL and we have maid User Attribute Value as text without any problem. However, this is only for PostgreSQL.
